### PR TITLE
Fixes bug 1271281 - Hide API base classes from public eyes.

### PR DESCRIPTION
**Pull request by adngdb**
What do you think about this?

### CODE REVIEW

File: webapp-django/crashstats/api/tests/test_views.py [hunk 1]
```diff
@@ -86,6 +86,15 @@ def test_invalid_url(self):
         response = self.client.get(url)
         eq_(response.status_code, 404)
 
+    def test_base_classes_raise_bad_request(self):
+        url = reverse('api:model_wrapper', args=('SocorroMiddleware',))
+        response = self.client.get(url)
+        eq_(response.status_code, 400)
```
Comment (peterbe) [inline webapp-django/crashstats/api/tests/test_views.py]: Should be 404. 400 is when the URL is valid but there's something wrong with the parameters. 
Equally, if you do `curl .../api/cm49b7aktgja7v` that should definitely be a 404.

File: webapp-django/crashstats/api/views.py [hunk 2]
```diff
@@ -223,6 +223,16 @@ def model_wrapper(request, model_name):
     else:
         raise http.Http404('no model called `%s`' % model_name)
 
+    if (
+        not issubclass(model, models.SocorroMiddleware) or
+        model is models.SocorroMiddleware or
+        model is crashstats.supersearch.models.ESSocorroMiddleware
+    ):
```
Comment (peterbe) [inline webapp-django/crashstats/api/views.py]: Please turn this into a function and re-use that around like 433.

**Comment by peterbe** [comment]
r+ with nits but I wouldn't mind taking a second last look.

**Comment by adngdb** [comment]
@peterbe edited!

**Pull request merged**